modules/lotamePanoramaIdSystem.js

```diff
@@ -0,0 +1,241 @@
+    } else if (storedValueExp) {
+      if ((new Date(storedValueExp)).getTime() - Date.now() > 0) {
+        value = storage.getDataFromLocalStorage(key, undefined);
+      }
+    }
+  }
+  return value;
+}
+
+/**
+ * Save a key/value pair to the browser cache (cookies and local storage)
+ * @param {String} key
+ * @param {String} value
+ * @param {Number} expirationTimestamp
+ */
+function saveLotameCache(
+  key,
+  value,
+  expirationTimestamp = utils.timestamp() + DAYS_TO_CACHE * DAY_MS
+) {
+  if (key && value) {
+    let expirationDate = new Date(expirationTimestamp).toUTCString();
+    if (storage.cookiesAreEnabled()) {
+      storage.setCookie(
+        key,
+        value,
+        expirationDate,
+        'Lax',
+        undefined,
+        undefined
+      );
+    }
+    if (storage.hasLocalStorage()) {
+      storage.setDataInLocalStorage(`${key}_exp`, expirationDate, undefined);
+      storage.setDataInLocalStorage(key, value, undefined);
+    }
+  }
+}
+
+/**
+ * Retrieve all the cached values from cookies and/or local storage
+ */
+function getLotameLocalCache() {
+  let cache = {
+    data: getFromStorage(KEY_ID),
+    expiryTimestampMs: 0,
+  };
+
+  try {
+    const rawExpiry = getFromStorage(KEY_EXPIRY);
+    if (utils.isStr(rawExpiry)) {
+      cache.expiryTimestampMs = parseInt(rawExpiry, 0);
+    }
+  } catch (error) {
+    utils.logError(error);
+  }
+
+  return cache;
+}
+
+/**
+ * Clear a cached value from cookies and local storage
+ * @param {String} key
+ */
+function clearLotameCache(key) {
+  if (key) {
+    if (storage.cookiesAreEnabled()) {
+      let expirationDate = new Date(0).toUTCString();
+      storage.setCookie(key, '', expirationDate, 'Lax', undefined, undefined);
+    }
+    if (storage.hasLocalStorage()) {
+      storage.removeDataFromLocalStorage(key, undefined);
+    }
+  }
+}
+/** @type {Submodule} */
+export const lotamePanoramaIdSubmodule = {
+
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+
+  /**
+   * Decode the stored id value for passing to bid requests
+   * @function decode
+   * @param {(Object|string)} value
+   * @returns {(Object|undefined)}
+   */
+  decode(value, configParams) {
+    return utils.isStr(value) ? { 'lotamePanoramaId': value } : undefined;
+  },
+
+  /**
+   * Retrieve the Lotame Panorama Id
+   * @function
+   * @param {SubmoduleParams} [configParams]
+   * @param {ConsentData} [consentData]
+   * @param {(Object|undefined)} cacheIdObj
+   * @returns {IdResponse|undefined}
+   */
+  getId(configParams, consentData, cacheIdObj) {
+    let localCache = getLotameLocalCache();
+
+    let refreshNeeded = false;
+    if (!utils.isStr(localCache.data)) {
+      refreshNeeded = Date.now() > localCache.expiryTimestampMs;
+    }
+
+    if (!refreshNeeded) {
+      return {
+        id: localCache.data
+      };
+    }
+
+    const storedUserId = getProfileId();
+    const hasGdprData =
```

Comment:
I think we should pass the gdprApplies value to tse as well if it is present, btw to make it parallel with bcp.

Reply:
So should I pass it in as `&gdpr_applies=[true|false]` then?

Reply:
yes - and if the value is unknown, don't pass it in at all